docs/spec/tla/ibc-core/ICS04PacketHandlers.tla

```diff
@@ -164,19 +164,19 @@ WritePacketCommitment(chain, packet) ==
                 !.connectionEnd.channelEnd.nextSendSeq = channelEnd.nextSendSeq + 1
               ]
          \* otherwise, do not update the chain store
-         ELSE chain
-    ELSE IF \* if the channel is unordered, 
-            \* add a packet committment in the chain store
-            /\ channelEnd.order = "UNORDERED"
-         THEN [chain EXCEPT 
-                !.packetCommitments =  
-                    chain.packetCommitments \union {[portID |-> packet.srcPortID,
-                                                     channelID |-> packet.srcChannelID,
-                                                     sequence |-> packet.sequence,
-                                                     timeoutHeight |-> packet.timeoutHeight]}
-              ]
-         \* otherwise, do not update the chain store
-         ELSE chain
+         ELSE IF \* if the channel is unordered, 
+                 \* add a packet committment in the chain store
+                 /\ channelEnd.order = "UNORDERED"
+              THEN [chain EXCEPT 
+                    !.packetCommitments =  
+                        chain.packetCommitments \union {[portID |-> packet.srcPortID,
+                                                         channelID |-> packet.srcChannelID,
+                                                         sequence |-> packet.sequence,
+                                                         timeoutHeight |-> packet.timeoutHeight]}
+                   ]
+              \* otherwise, do not update the chain store
+              ELSE chain
+    ELSE chain              
```

Comment:
It's not clear to me why this change is required. The `TypeOK` invariant should not hold if the `channelEnd.order \not \in {"ORDERED", "UNORDERED"}`.

Reply:
This change was added because the `ELSE IF channelEnd.order = "UNORDERED"` was wrongly aligned -- a packet commitment on an unordered channel was written even if the conditions on lines 143-153 did not hold. This in fact means that a packet is sent on an uninitialized or closed channel, which should not be the case. 

The line 178 can be the outcome of `WritePacketCommitment` in case the channel is ordered, but the sequence numbers are not correct. This is not a violation of the type invariant, but of the property which checks if packets are sent in order on ordered channels. We currently don't have any properties/invariants for the packet transmission in the IBC spec. We should address this in a separate PR.

Reply:
Oh, I see. Thanks for explaining.